core/state_processor.go

```diff
@@ -17,11 +17,15 @@ var (
 )
 
 type StateProcessor struct {
-	bc *BlockChain
+	config *ChainConfig
+	bc     *BlockChain
 }
 
-func NewStateProcessor(bc *BlockChain) *StateProcessor {
-	return &StateProcessor{bc}
+func NewStateProcessor(config *ChainConfig, bc *BlockChain) *StateProcessor {
```

Comment:
As discussed, the config field for this method can be retrieved from the blockchain, so duplicating the data would just open up the potential for strange programming errors when the two mismatch. Please from config and fetch it from blockchain.